tests/test_catx.py

```diff
@@ -43,78 +47,167 @@ def catx(mlp_builder: NetworkBuilder, request: pytest.FixtureRequest = None) ->
 
 
 @pytest.fixture
-def create_forward_fn_mock(mocker: MockerFixture) -> MagicMock:
-    mk = mocker.patch("catx.catx.CATX._create_forward_fn")
-    return mk  # type: ignore
+def catx_init(
+    catx: CATX,
+    jax_observations: JaxObservations,
+    epsilon: float,
+    key: PRNGKey,
+) -> Tuple[CATX, CATXState]:
+
+    state = catx.init(obs=jax_observations, epsilon=epsilon, key=key)
+
+    return catx, state
 
 
 @pytest.fixture
-def create_forward_single_depth_fns_mock(mocker: MockerFixture) -> MagicMock:
-    mk = mocker.patch("catx.catx.CATX._create_forward_single_depth_fns")
-    mk.return_value = MagicMock(), MagicMock()
+def catx_with_dropout_extras(
+    catx_network_with_dropout_extras: Type[CATXHaikuNetwork],
+    request: pytest.FixtureRequest = None,
+) -> CATX:
+    if not request:
+        action_min = 0.0
+        action_max = 1.0
+    else:
+        action_min = request.param[0]
+        action_max = request.param[1]
 
-    return mk  # type: ignore
+    catx = CATX(
+        catx_network=catx_network_with_dropout_extras,
+        optimizer=optax.adam(learning_rate=0.01),
+        discretization_parameter=4,
+        bandwidth=1.5 / 4,
+        action_min=action_min,
+        action_max=action_max,
+    )
+
```

Comment:
we can probably create a function that takes in the catx_network as parameter as well as the other fixture parameters and returns the CATX object instanciated. Since the fixtures catx and catx_with_dropout_extras differ only by the class used for the catx network.